### PR TITLE
add s3 status reporting for integration test builds

### PR DESCRIPTION
**Pull request by jrhizor**
## What
- Report status of `master` builds to S3.
- Create summary files that can be used as a shields.io badge and as a listing to find failing bugs for debugging purposes.

Reading order:
- `init.sh` - shouldn't ever be run, just a record of what I did
- `test-command.yml` - how we launch the reporting
- `report.sh` - all of the logic for reporting

**Please look at my later comment with examples from two of the branch builds here.**

I'll add badges everywhere and the actual "dashboard view" in a separate PR. I'll also remove the legacy kv-store / python dashboard in a separate PR.

I've manually tested each case of the displays (all failures, all successes, last one passing with at least one failure, last one failing with at least one success).

### CODE REVIEW

File: tools/status/report.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+HTML="$HTML_TOP $HTML_TABLE $HTML_BOTTOM"
+BADGE="{ \"schemaVersion\": 1, \"label\": \"\", \"labelColor\": \"lightblue\", \"message\": \"$message\", \"color\": \"$color\", \"cacheSeconds\": 300, \"logoSvg\": \"<svg version=\\\"1.0\\\" xmlns=\\\"http://www.w3.org/2000/svg\\\"\\n width=\\\"32.000000pt\\\" height=\\\"32.000000pt\\\" viewBox=\\\"0 0 32.000000 32.000000\\\"\\n preserveAspectRatio=\\\"xMidYMid meet\\\">\\n\\n<g transform=\\\"translate(0.000000,32.000000) scale(0.100000,-0.100000)\\\"\\nfill=\\\"#000000\\\" stroke=\\\"none\\\">\\n<path d=\\\"M136 279 c-28 -22 -111 -157 -102 -166 8 -8 34 16 41 38 8 23 21 25\\n29 3 3 -8 -6 -35 -20 -60 -18 -31 -22 -44 -12 -44 20 0 72 90 59 103 -6 6 -11\\n27 -11 47 0 77 89 103 137 41 18 -23 16 -62 -5 -96 -66 -109 -74 -125 -59\\n-125 24 0 97 140 97 185 0 78 -92 123 -154 74z\\\"/>\\n<path d=\\\"M168 219 c-22 -13 -23 -37 -2 -61 12 -12 14 -22 7 -30 -5 -7 -22 -34\\n-37 -60 -20 -36 -23 -48 -12 -48 13 0 106 147 106 169 0 11 -28 41 -38 41 -4\\n0 -15 -5 -24 -11z m32 -34 c0 -8 -4 -15 -10 -15 -5 0 -10 7 -10 15 0 8 5 15\\n10 15 6 0 10 -7 10 -15z\\\"/>\\n</g>\\n</svg>\\n\" }"
```
Comment (jrhizor) [inline tools/status/report.sh:86]: There's a minimum of 5min caching for shields.io unfortunately.

**Comment by jrhizor** [comment]
# Examples

## source-exchangeratesapi-singer

Markdown code:
```
[![source-exchangeratesapi-singer](https://img.shields.io/endpoint?url=https%3A%2F%2Fstatus-api.airbyte.io%2Ftests%2Fsummary%2Fsource-exchangeratesapi-singer%2Fbadge.json)](https://status-api.airbyte.io/tests/summary/source-exchangeratesapi-singer)
```

Result (remember to try clicking on it!):

[source-exchangeratesapi-singer](https://status-api.airbyte.io/tests/summary/source-exchangeratesapi-singer)

---

## source-stripe-singer

Markdown code:
```
[![source-stripe-singer](https://img.shields.io/endpoint?url=https%3A%2F%2Fstatus-api.airbyte.io%2Ftests%2Fsummary%2Fsource-stripe-singer%2Fbadge.json)](https://status-api.airbyte.io/tests/summary/source-stripe-singer)
```

Result (remember to try clicking on it!):

[source-stripe-singer](https://status-api.airbyte.io/tests/summary/source-stripe-singer)

File: tools/status/report.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  outcome=$(echo "$line" | jq -r '.outcome')
+  if [ "$outcome" = "success" ]; then
+    successes=$((successes+1))
+    outcome_output="<span style=\"color:green;\">&#10004; $outcome</span>"
+  else
+    failures=$((failures+1))
+    outcome_output="<span style=\"color:red;\">&#10008; $outcome</span>"
+  fi
+  LINK=$(echo "$line" | jq -r '.link')
+  HTML_TABLE_ROWS="<tr><td>$(date -d @"$(echo "$file" | cut -f 1 -d '.')")</td><td>$outcome_output</td><td><a href=\"$LINK\">$LINK</a></td></tr>$HTML_TABLE_ROWS"
```
Comment (jrhizor) [inline tools/status/report.sh:57]: `date -r` works on Mac but doesn't work on Github Actions. `date -d @` works on Github Actions but not on Mac. This is the only thing that prevents this script from being manually run locally.

Comment (cgardens) [reply tools/status/report.sh:57]: if that's the case can you just do a switch based on how the OS? it's ugly but i think it's important that it runs in both places.

**Review by cgardens — APPROVED** [review]
nice! i might recommend breaking some of report.sh into separate methods so it's easier to read. but also i get it, it's bash.

**Review by sherifnada — APPROVED** [review]

**Review by michel-tricot — APPROVED** [review]

**Pull request merged**